packages/server-common/test/handlers/calendar/unsubscribe-calendars.test.ts

```diff
@@ -0,0 +1,190 @@
+    unsubscribeMock,
+    logger,
+  );
+  const invokeHandler = () =>
+    unsubscribeCalendarsHandler(
+      createEventBridgeScheduledEventMock(),
+      createHandlerContext(),
+    );
+
+  const getCalendarId = (id: string) => `cms:${id}`;
+
+  const unsubscribeCalendarsHandlerWithIdFunction =
+    unsubscribeCalendarsHandlerFactory(
+      calendarDataProviderMock,
+      unsubscribeMock,
+      logger,
+      getCalendarId,
+    );
+  const invokeHandlerWithIdFunction = () =>
+    unsubscribeCalendarsHandlerWithIdFunction(
+      createEventBridgeScheduledEventMock(),
+      createHandlerContext(),
+    );
+
+  const fakeNow = 1614697798681;
+  const realDate = Date.now.bind(Date);
+
+  beforeAll(() => {
+    Settings.now = () => fakeNow;
+  });
+
+  afterAll(() => {
+    Settings.now = realDate;
+  });
+
+  afterEach(() => {
+    jest.clearAllMocks();
+    jest.resetAllMocks();
+  });
+
+  test('Should get the list of calendars expiring within 24h', async () => {
+    calendarDataProviderMock.fetch.mockResolvedValueOnce({
+      total: 0,
+      items: [],
+    });
+
+    await invokeHandler();
+
+    expect(calendarDataProviderMock.fetch).toHaveBeenCalledWith({
+      maxExpiration: fakeNow + 24 * 60 * 60 * 1000,
+    });
+  });
+
+  test('Should get the list of calendars and unsubscribe each of them', async () => {
+    const calendarDataObject1 = {
+      ...getCalendarDataObject(),
+      channelId: undefined,
+    };
+    const calendarDataObject2 = {
+      ...getCalendarDataObject(),
+      id: 'uuid2',
+      resourceId: 'resource-id-2',
+      channelId: undefined,
+    };
+
+    calendarDataProviderMock.fetch.mockResolvedValueOnce({
+      items: [calendarDataObject1, calendarDataObject2],
+      total: 2,
+    });
+
+    await invokeHandler();
+
+    expect(unsubscribeMock).toHaveBeenCalledWith(
+      calendarDataObject1.resourceId,
+      calendarDataObject1.id,
+    );
+    expect(unsubscribeMock).toHaveBeenCalledWith(
+      calendarDataObject2.resourceId,
+      calendarDataObject2.id,
+    );
+  });
+
+  test('Should use `getCalendarId` function if provided when unsubscribing', async () => {
+    const calendarDataObject1 = {
+      ...getCalendarDataObject(),
+      channelId: undefined,
+    };
+    const calendarDataObject2 = {
+      ...getCalendarDataObject(),
+      id: 'uuid2',
+      resourceId: 'resource-id-2',
+      channelId: undefined,
+    };
+
+    calendarDataProviderMock.fetch.mockResolvedValueOnce({
+      items: [calendarDataObject1, calendarDataObject2],
+      total: 2,
+    });
+
+    await invokeHandlerWithIdFunction();
+    expect(unsubscribeMock).toHaveBeenCalledWith(
+      calendarDataObject1.resourceId,
+      `cms:${calendarDataObject1.id}`,
+    );
+    expect(unsubscribeMock).toHaveBeenCalledWith(
+      calendarDataObject2.resourceId,
+      `cms:${calendarDataObject2.id}`,
+    );
+  });
+
+  test('Should remove the resourceId after unsubscribing', async () => {
```

Comment:
I actually wonder if we should only drop the resourceId when unsubscribing or should we just drop all the metadata, what do you think?

Reply:
yeah, I think we can drop all metadata 👍